client/login/controller.js

```diff
@@ -409,6 +409,155 @@ export async function jetpackGoogleAuthCallback( context, next ) {
 	return next();
 }
 
+export async function jetpackAppleAuth( context, next ) {
+	const { query } = context;
+	const redirectUri = `https://${
+		config.isEnabled( 'oauth' ) ? window.location.host : 'wordpress.com'
+	}${ loginPath( { socialService: 'apple' } ) }`;
+
+	try {
+		// Get authorization nonce for security
+		let nonce;
+		if ( config.isEnabled( 'oauth' ) ) {
+			// Use the API to get a real nonce in production
+			const response = await wpcomRequest( {
+				path: '/generate-authorization-nonce',
+				apiNamespace: 'wpcom/v2',
+				method: 'GET',
+			} );
+			nonce = response.nonce;
+		} else {
+			// In dev environment, use a random string as nonce
+			nonce = Math.random().toString( 36 ).substring( 2, 15 );
+		}
+
+		// Create state object with relevant data
+		const stateObject = {
+			is_jetpack: true,
+			oauth2State: nonce,
+			// Allow just redirect_to to be passed in the query params
+			queryString: `redirect_to=${ query?.redirect_to || '/' }`,
+		};
+
+		// Store nonce in sessionStorage for validation on callback
+		window.sessionStorage.setItem( 'siwa_state', nonce );
+
+		// Load Apple client if not already loaded
+		const appleClientUrl =
+			'https://appleid.cdn-apple.com/appleauth/static/jsapi/appleid/1/en_US/appleid.auth.js';
+
+		if ( ! window.AppleID ) {
+			await loadScript( appleClientUrl );
+			if ( ! window.AppleID ) {
+				throw new Error( 'Failed to load Apple Authentication Services API' );
+			}
+		}
```

Comment:
We've been using a similar approach to load the script of the other clients as well, WDYT about creating a method to handle that to make the code more readable?

Reply:
Good point, I will make all scripts more modular once we get all of them working well.